sympy/vector/operators.py

```diff
@@ -0,0 +1,184 @@
+from sympy.core import Expr
+from sympy.vector.coordsysrect import CoordSysCartesian
+from sympy.vector.vector import Vector
+from sympy.vector.deloperator import Del
+
+
+class Gradient(Expr):
+    """
+    Represents unevaluated Gradient.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> s = R.x*R.y*R.z
+    >>> Gradient(s, R)
+    Gradient(R.x*R.y*R.z, R)
+
+    """
+
+    def __new__(cls, scalar, coord_sys):
+        obj = Expr.__new__(cls, scalar, coord_sys)
+        obj._scalar = scalar
+        obj._coord_sys = coord_sys
+        return obj
+
+    def doit(self):
+        return Del(self._coord_sys).gradient(self._scalar).doit()
+
+
+class Divergence(Expr):
+    """
+    Represents unevaluated Divergence.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> Divergence(v, R)
+    Divergence(R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k, R)
+
+    """
+
+    def __new__(cls, vect, coord_sys):
+        obj = Expr.__new__(cls, vect, coord_sys)
+        obj._vect = vect
+        obj._coord_sys = coord_sys
+        return obj
+
+    def doit(self):
+        return Del(self._coord_sys).dot(self._vect).doit()
+
+
+class Curl(Expr):
+    """
+    Represents unevaluated Curl.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Curl
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> Curl(v, R)
+    Curl(R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k, R)
+
+    """
+
+    def __new__(cls, vect, coord_sys):
+        obj = Expr.__new__(cls, vect, coord_sys)
+        obj._vect = vect
+        obj._coord_sys = coord_sys
+        return obj
+
+    def doit(self):
+        return Del(self._coord_sys).cross(self._vect).doit()
+
+
+def gradient(scalar, coord_sys=None):
+    """
+    Returns the vector gradient of a scalar field computed wrt the
+    base scalars of the given coordinate system.
+
+    Parameters
+    ==========
+
+    scalar : SymPy Expr
+        The scalar field to compute the gradient of
+
+    coord_sys : CoordSysCartesian
+        The coordinate system to calculate the gradient in
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, gradient
+    >>> R = CoordSysCartesian('R')
+    >>> s1 = R.x*R.y*R.z
+    >>> gradient(s1, R)
+    R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> s2 = 5*R.x**2*R.z
+    >>> gradient(s2, R)
+    10*R.x*R.z*R.i + 5*R.x**2*R.k
+
+    """
+
+    if coord_sys is None:
+        coord_sys = list(scalar.atoms(CoordSysCartesian))[0]
```

Comment:
Put this in a private function, like `_get_coord_sys_from_expr(...)`, this code is shared by all operators.